tests/docker-compose-hdfs-overlay.yaml

```diff
@@ -0,0 +1,250 @@
+services:
+  server:
+    hostname: server
+    environment:
+      - KRB5CCNAME=FILE:/tickets/krb5cc_0
+      - KRB5_CONFIG=/etc/krb5.conf
+      - HADOOP_CONF_DIR=/etc/hadoop
+      - KRB5_TRACE=/dev/stderr
+      - RUST_LOG=debug,hdfs_native=trace
+    volumes:
+      - ./kerberos-tickets:/tickets:ro
+      - ./hdfs/krb5.conf:/etc/krb5.conf:ro
+      - ./hdfs/hadoop-config/core-site.xml:/etc/hadoop/core-site.xml:ro
+      - ./hdfs/hadoop-config/hdfs-site.xml:/etc/hadoop/hdfs-site.xml:ro
+    depends_on:
+      - kerberos-sidecar
+    networks:
+      iceberg_rest_tests:
+
+  # Kerberos KDC (Key Distribution Center)
+  keycloak:
+    ports:
+      - "8080:8080"
+      - "88:88/tcp"
+      - "88:88/udp"
+  kdc:
+    image: sequenceiq/kerberos:latest
+    hostname: kdc.example.com
+    entrypoint: [ "/bin/bash", "-c" ]
+    command: |
+      "
+      # Stop services to reconfigure
+      /etc/rc.d/init.d/krb5kdc stop
+      /etc/rc.d/init.d/kadmin stop
+
+      # Delete old database
+      kdb5_util destroy -f -r NODE.DC1.CONSUL
+
+      # Remove any existing keytabs
+      rm -f /keytabs/*.keytab
+      rm -f /keytabs/ready
+
+      # Update KDC config
+      sed -i 's/NODE.DC1.CONSUL/EXAMPLE.COM/g' /var/kerberos/krb5kdc/kdc.conf
+
+      # Initialize new database
+      mkdir -p /var/log/kerberos
+      kdb5_util -P masterkey -r EXAMPLE.COM create -s
+
+      # Create principals
+      kadmin.local -q 'addprinc -pw admin admin/admin'
+      echo '*/admin@EXAMPLE.COM *' > /var/kerberos/krb5kdc/kadm5.acl
+      
+      # Service principals
+      kadmin.local -q 'addprinc -randkey hdfs/namenode@EXAMPLE.COM'
+      kadmin.local -q 'addprinc -randkey HTTP/namenode@EXAMPLE.COM'
+      kadmin.local -q 'addprinc -randkey hdfs/datanode@EXAMPLE.COM'
+      
+      # Client principal
+      kadmin.local -q 'addprinc -randkey hdfs/server@EXAMPLE.COM'
+      
+      # Create keytabs
+      kadmin.local -q 'ktadd -k /keytabs/dn.keytab hdfs/datanode@EXAMPLE.COM'
+      kadmin.local -q 'ktadd -k /keytabs/app.keytab hdfs/server@EXAMPLE.COM'
+      kadmin.local -q 'ktadd -k /keytabs/nn.keytab hdfs/namenode@EXAMPLE.COM HTTP/namenode@EXAMPLE.COM'
+      # Create ready file explicitly
+      echo 'Keytab created' > /keytabs/ready
+      chmod 644 /keytabs/ready
+
+      # Start services
+      /etc/rc.d/init.d/krb5kdc start
+      /etc/rc.d/init.d/kadmin start
+
+      # Keep container running
+      tail -F /var/log/kerberos/krb5kdc.log
+      "
+    volumes:
+      - ./keytabs:/keytabs
+      - ./hdfs/krb5.conf:/etc/krb5.conf:ro
+    networks:
+      iceberg_rest_tests:
+        aliases:
+          - kdc.example.com
+  # HDFS NameNode with Kerberos
+  namenode:
+    image: apache/hadoop:3
+    hostname: namenode
+    container_name: namenode
+    user: root
```

Comment:
may be no longer needed since we have a ssl secured cluster

Reply:
still required